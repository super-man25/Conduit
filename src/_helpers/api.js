import { baseURL } from '_constants';
import { stringify } from 'querystring';

/**
 * API Error - compatible with instanceof
 */
export class ApiError {
  constructor(code, message, body) {
    this.code = code;
    this.message = message;
    this.body = body;
  }
}
Object.setPrototypeOf(ApiError, Error);

/**
 * Handle response from a request (expect json)
 * @private
 *
 * @param {Object} response - Fetch response object
 */
export function handleResponse(response) {
  const statusCode = response.status;
  if (statusCode < 400) {
    return response.json();
  }

  return response
    .json()
    .catch(() => null)
    .then((body) => {
      throw new ApiError(statusCode, response.statusText, body);
    });
}

/**
 * Generic request
 *
 * @param {string} path - request path (no leading "/")
 * @param {Object} opts - options passed on to the fetch request
 */
export function request(path, opts = {}) {
  return fetch(`${baseURL}/${path}`, {
    credentials: 'include',
    mode: 'cors',
    ...opts
  }).then(handleResponse);
}

/**
 * GET request
 *
 * @param {string} path - request path (no leading "/")
 * @param {Object} params - request params in object form
 * @param {Object} opts - options passed on to the fetch request
 */
export function get(path, params = {}, opts = {}) {
  const search = stringify(params);
  return request(`${path}?${search}`, {
    method: 'GET',
    ...opts
  });
}

/**
 * POST request
 *
 * @param {string} path - request path (no leading "/")
 * @param {Object} body - requesty body
 * @param {Object} opts - options passed on to the fetch request
 */
export function post(path, body = {}, opts = {}) {
  return request(path, {
    method: 'POST',
    body: JSON.stringify(body),
    ...opts
  });
}

/**
 * PUT request
 *
 * @param {string} path - request path (no leading "/")
 * @param {Object} body - requesty body
 * @param {Object} opts - options passed on to the fetch request
 */
export function put(path, body = {}, opts = {}) {
  return request(path, {
    method: 'PUT',
    body: JSON.stringify(body),
    ...opts
  });
}

/**
 * DELETE request
 *
 * @param {string} path - request path (no leading "/")
 * @param {Object} params - request params in object form
 * @param {Object} opts - options passed on to the fetch request
 */
export function del(path, params = {}, opts = {}) {
  const search = stringify(params);
  return request(`${path}?${search}`, {
    method: 'DELETE',
    ...opts
  });
}
