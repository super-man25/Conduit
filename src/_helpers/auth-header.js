export function authHeader() {
  // return authorization header with jwt token
  // const user = JSON.parse(localStorage.getItem('user'));
  const user = {};

  if (user && user.token) {
    return { 'Authorization': `Bearer ${ user.token}` };
  }
  return {};

}
