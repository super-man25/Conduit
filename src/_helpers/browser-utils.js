import { mobileBreakpoint } from '_constants';

export const isMobileDevice =
  document.body && document.body.clientWidth <= mobileBreakpoint;
