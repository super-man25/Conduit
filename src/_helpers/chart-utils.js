// @flow
import React from 'react';
import {
  isAfter,
  isBefore,
  differenceInHours,
  format,
  differenceInCalendarDays,
} from 'date-fns';

import {
  READABLE_TIME_FORMAT,
  READABLE_MONTH_FORMAT,
  CONCISE_READABLE_DATE_FORMAT,
  ABBREVIATED_READABLE_DATE_FORMAT,
  READABLE_MONTH_YEAR_FORMAT,
  DAILY,
  END_OF_DAY,
  WEEKLY,
  BI_WEEKLY,
  MONTHLY,
  BUSINESS_WEEK,
  END_OF_MONTH,
  QUARTERLY,
} from '_constants';
import { formatDate, formatUSD, formatNumber } from './string-utils';
import { EventStat, EventStatInterval } from '_models';
import { MinorXAxisTick } from '_components/Charts/MinorXAxisTick';
import {
  MajorXAxisTick,
  MajorXAxisTickLine,
} from '_components/Charts/MajorXAxisTick';

export type ChartPoint = { x: Date, y: number };

type DataSet = {
  [string]: Array<ChartPoint>,
};

type TickProps = {
  tickTotal: number,
  tickFormat: (any) => string,
};

type RenderXAxisTicksProps = {
  interval: EventStatInterval,
  tickProps: {
    x: number,
    y: number,
    payload: { value: Date },
    index: number,
  },
  timeZone: Object,
  dataLength: number,
};

export function dayFormat(t: Date): string {
  return format(new Date(t), 'MM/DD');
}

export function dayTimeFormat(t: Date): string {
  return format(new Date(t), 'MM/DD HH:MM');
}

export function periodicTooltip(stat: EventStat) {
  const periodicInventory = stat.isProjected
    ? stat.projectedPeriodicInventory
    : stat.periodicInventory;
  const periodicRevenue = stat.isProjected
    ? stat.projectedPeriodicRevenue
    : stat.periodicRevenue;
  const avgTicketPrice = stat.isProjected
    ? stat.projectedPeriodicRevenue / (stat.projectedPeriodicInventory * -1)
    : stat.periodicRevenue / (stat.periodicInventory * -1);

  return {
    periodicInventory: formatNumber(periodicInventory),
    periodicRevenue: formatUSD(periodicRevenue),
    avgTicketPrice: formatUSD(avgTicketPrice),
  };
}

export function cumulativeTooltip(stat: EventStat, totalInventory: number) {
  const inventory = stat.isProjected ? stat.projectedInventory : stat.inventory;
  const revenue = stat.isProjected ? stat.projectedRevenue : stat.revenue;
  const avgTicketPrice = stat.isProjected
    ? stat.projectedRevenue / (totalInventory - stat.projectedInventory)
    : stat.revenue / Math.abs(stat.soldInventory);

  return {
    inventory: formatNumber(inventory),
    revenue: formatUSD(revenue),
    avgTicketPrice: formatUSD(avgTicketPrice),
  };
}

export function getAxisTickOptions(dataset: DataSet = {}): TickProps {
  let firstDay;
  let lastDay;

  for (const key of Object.keys(dataset)) {
    for (const point of dataset[key]) {
      const { x: date } = point;
      if (!firstDay) firstDay = date;
      if (!lastDay) lastDay = date;

      if (isAfter(date, lastDay)) {
        lastDay = date;
      } else if (isBefore(date, firstDay)) {
        firstDay = date;
      }
    }
  }

  if (!firstDay || !lastDay) {
    return { tickTotal: 10, tickFormat: dayFormat };
  }

  const totalDays = differenceInCalendarDays(lastDay, firstDay);

  if (totalDays <= 1) {
    return {
      tickTotal: Math.min(differenceInHours(lastDay, firstDay), 6),
      tickFormat: dayTimeFormat,
    };
  } else if (totalDays) {
    return {
      tickTotal: Math.min(totalDays, END_OF_MONTH),
      tickFormat: dayFormat,
    };
  }

  // Fallback
  return { tickTotal: 10, tickFormat: dayFormat };
}

export function getChartRange(dataset: DataSet = {}): [number, number] {
  let max = -Infinity;
  let min = 0;
  for (const key of Object.keys(dataset)) {
    for (const point of dataset[key]) {
      const { y } = point;
      max = Math.max(max, y);
      min = Math.min(min, y);
    }
  }

  return [min, max];
}

/**
 * renderMinorXAxisTicks() has custom logic on how to breakdown
 * the ticks shown in the charts based on defined constants valuable to the
 * business.
 */
export function renderMinorXAxisTicks({
  interval,
  tickProps,
  dataLength,
  timeZone,
}: RenderXAxisTicksProps) {
  const {
    x,
    y,
    payload: { value },
  } = tickProps;
  let ticksToRender = [];
  let formattedValue = '';

  if (interval === 'Hours') {
    const dayRange = Math.ceil(dataLength / END_OF_DAY);

    switch (true) {
      case dayRange <= DAILY:
        // The ticks will render for the following hours:
        // 12:00am - 4:00am - 8:00am - 12:00pm - 4:00pm - 8:00pm
        ticksToRender = Array.from({ length: 6 }, (_, i) => `${i * 4}`);
        formattedValue = formatDate(value, READABLE_TIME_FORMAT, timeZone);
        break;
      case dayRange <= BUSINESS_WEEK:
        // 12:00am - 12:00pm
        ticksToRender = Array.from({ length: 2 }, (_, i) => `${i * 12}`);
        formattedValue = formatDate(value, READABLE_TIME_FORMAT, timeZone);
        break;
      case dayRange <= WEEKLY:
        // 12:00am
        ticksToRender = ['0'];
        formattedValue = formatDate(
          value,
          CONCISE_READABLE_DATE_FORMAT,
          timeZone
        );
        break;
      case dayRange <= BI_WEEKLY:
        // 12:00am
        ticksToRender = ['0'];
        formattedValue = formatDate(
          value,
          ABBREVIATED_READABLE_DATE_FORMAT,
          timeZone
        );
        break;
      default:
        // 12:00am
        ticksToRender = ['0'];
        formattedValue = formatDate(
          value,
          CONCISE_READABLE_DATE_FORMAT,
          timeZone
        );
        break;
    }

    const currentHour = formatDate(value, 'H', timeZone);
    if (ticksToRender.includes(currentHour))
      return <MinorXAxisTick x={x} y={y} value={formattedValue} />;
  }

  if (interval === 'Days') {
    const monthRange = Math.ceil(dataLength / END_OF_MONTH);
    switch (true) {
      case monthRange <= MONTHLY:
        // The ticks will render for the following days:
        // 1st - 8th - 15th - 22nd
        const weeksToFit = Math.floor(dataLength / WEEKLY);
        ticksToRender = Array.from(
          { length: weeksToFit },
          (_, i) => `${i * WEEKLY + 1}`
        );
        formattedValue = formatDate(
          value,
          CONCISE_READABLE_DATE_FORMAT,
          timeZone
        );
        break;
      case monthRange <= QUARTERLY:
        // 1st - 15th
        ticksToRender = Array.from(
          { length: 2 },
          (_, i) => `${i * BI_WEEKLY + 1}`
        );
        formattedValue = formatDate(
          value,
          CONCISE_READABLE_DATE_FORMAT,
          timeZone
        );
        break;
      default:
        // 1st
        ticksToRender = ['1'];
        formattedValue = formatDate(value, READABLE_MONTH_FORMAT, timeZone);
        break;
    }

    const currentDay = formatDate(value, 'D', timeZone);
    if (ticksToRender.includes(currentDay))
      return <MinorXAxisTick x={x} y={y} value={formattedValue} />;
  }
}

/**
 * renderMajorXAxisTicks() has custom logic on how to breakdown
 * the ticks shown in the charts based on defined constants valuable to the
 * business.
 */
export function renderMajorXAxisTicks({
  interval,
  tickProps,
  dataLength,
  timeZone,
}: RenderXAxisTicksProps) {
  const {
    x,
    y,
    payload: { value },
  } = tickProps;
  let ticksToRender = [];
  let tickLinesToRender = [];
  let formattedValue = '';

  if (interval === 'Hours') {
    const dayRange = Math.ceil(dataLength / END_OF_DAY);

    if (dayRange <= BUSINESS_WEEK) {
      ticksToRender = ['12']; // tick is placed in middle of day
      tickLinesToRender = ['0']; // tick line is placed at start of day
      formattedValue = formatDate(
        value,
        CONCISE_READABLE_DATE_FORMAT,
        timeZone
      );
    }

    const currentHour = formatDate(value, 'H', timeZone);
    if (ticksToRender.includes(currentHour))
      return <MajorXAxisTick x={x} y={y} value={formattedValue} />;

    if (tickLinesToRender.includes(currentHour))
      return <MajorXAxisTickLine x={x} y={y} value={formattedValue} />;
  }

  if (interval === 'Days') {
    const monthRange = Math.ceil(dataLength / END_OF_MONTH);

    if (monthRange === MONTHLY) {
      ticksToRender = [`${Math.floor(dataLength / 2)}`]; // tick is placed in middle of days shown
      tickLinesToRender = ['1']; // tick line is placed at start of month
      formattedValue = formatDate(value, READABLE_MONTH_YEAR_FORMAT, timeZone);
    } else if (monthRange <= QUARTERLY) {
      ticksToRender = ['15']; // tick is placed in middle of month
      tickLinesToRender = ['1']; // tick line is placed at start of month
      formattedValue = formatDate(value, READABLE_MONTH_YEAR_FORMAT, timeZone);
    }

    const currentDay = formatDate(value, 'D', timeZone);
    if (ticksToRender.includes(currentDay))
      return <MajorXAxisTick x={x} y={y} value={formattedValue} />;

    if (tickLinesToRender.includes(currentDay))
      return <MajorXAxisTickLine x={x} y={y} value={formattedValue} />;
  }
}

/**
 * renderMobileXAxisTicks() has custom logic on how to breakdown
 * the ticks shown in the charts based on defined constants valuable to the
 * business.
 */
export function renderMobileXAxisTicks({
  interval,
  tickProps,
  dataLength,
  timeZone,
}: RenderXAxisTicksProps) {
  const {
    x,
    y,
    payload: { value },
  } = tickProps;
  let ticksToRender = [];
  let tickLinesToRender = [];
  let formattedValue = '';

  if (interval === 'Hours') {
    const dayRange = Math.ceil(dataLength / END_OF_DAY);

    if (dayRange <= BUSINESS_WEEK) {
      ticksToRender = ['12']; // tick is placed in middle of day
      tickLinesToRender = ['0']; // tick line is placed at start of day
      formattedValue = formatDate(
        value,
        CONCISE_READABLE_DATE_FORMAT,
        timeZone
      );
    } else {
      ticksToRender = ['12']; // tick is placed in middle of day
      tickLinesToRender = ['0']; // tick line is placed at start of day
      formattedValue = formatDate(value, 'ddd', timeZone);
    }

    const currentHour = formatDate(value, 'H', timeZone);
    if (ticksToRender.includes(currentHour))
      return <MajorXAxisTick x={x} y={y} value={formattedValue} />;

    if (tickLinesToRender.includes(currentHour))
      return <MajorXAxisTickLine x={x} y={y} value={formattedValue} />;
  }

  if (interval === 'Days') {
    const monthRange = Math.ceil(dataLength / END_OF_MONTH);

    if (monthRange === MONTHLY) {
      ticksToRender = [`${Math.floor(dataLength / 2)}`]; // tick is placed in middle of days shown
      tickLinesToRender = ['1']; // tick line is placed at start of month
      formattedValue = formatDate(value, 'MMMM', timeZone);
    } else if (monthRange <= QUARTERLY) {
      ticksToRender = ['15']; // tick is placed in middle of month
      tickLinesToRender = ['1']; // tick line is placed at start of month
      formattedValue = formatDate(value, 'MMMM', timeZone);
    } else {
      ticksToRender = ['15']; // tick is placed in middle of month
      tickLinesToRender = ['1']; // tick line is placed at start of month
      formattedValue = formatDate(value, 'MMM', timeZone);
    }

    const currentDay = formatDate(value, 'D', timeZone);
    if (ticksToRender.includes(currentDay))
      return <MajorXAxisTick x={x} y={y} value={formattedValue} />;

    if (tickLinesToRender.includes(currentDay))
      return <MajorXAxisTickLine x={x} y={y} value={formattedValue} />;
  }
}
