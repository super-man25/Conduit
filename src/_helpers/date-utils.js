//@flow
import { isBefore, subMinutes } from 'date-fns';
import { Event } from '_models';

/**
 * Check if the event timestamp passed more than an hour ago
 */
export function isPastEvent(event: Event) {
  const pastHour = subMinutes(new Date(), 60);
  return isBefore(event.timestamp, pastHour);
}
