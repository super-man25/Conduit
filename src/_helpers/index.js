export * from './browser-utils';
export * from './chart-utils';
export * from './date-utils';
export * from './history';
export * from './js-utils';
export * from './pricing-utils';
export * from './string-utils';
export * from './style-utils';
export * from './svg-utils';
