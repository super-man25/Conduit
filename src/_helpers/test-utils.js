/**
 * return a string with the ISO date as well as the format that was passed in.
 * This is an attempt to get around date-fns using the system timezone in CI
 *
 * @param {Date|Number} date
 * @param {String} format
 * @returns
 */
export function mockDateFnsFormat(date, format) {
  return `${new Date(date).toISOString()} @${format}`;
}

/**
 * return a boolean that checks if the timestamp is after today's date
 *
 * @param {Date|Number} timestamp
 * @param {Date} today
 */
export function mockIsAfter(timestamp, today) {
  return new Date(timestamp) > today;
}
