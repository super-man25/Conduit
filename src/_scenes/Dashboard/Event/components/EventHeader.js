// @flow
import * as React from 'react';
import { H1, H4, Flex, FlexItem, EDLink, P1, S1 } from '_components';
import { cssConstants } from '_constants';
import { format } from 'date-fns';
import { EDEvent } from '_models';
// import { formatNumber } from '_helpers/string-utils';

const DATE_FORMAT = 'dddd, MMMM Do, YYYY @ h:mmA';

const EventTitle = H1.extend`
  margin: 0;
  margin-bottom: 0.15rem;
`;

const EventDate = H4.extend`
  color: ${cssConstants.SECONDARY_BLUE};
  margin: 0;
  margin-bottom: 0.15rem;
`;

// const InventoryLink = S1.extend`
//   color: ${cssConstants.PRIMARY_LIGHT_BLUE};
//   text-decoration: underline;
// `;

// const ItalicS1 = S1.extend`
//   font-style: italic;
// `;

type Props = {
  event: EDEvent,
  availableInventory: number,
  totalInventory: number
};

// TODO: When Inventory list/Inventory totals are available we can add the commented out portions
export function EventHeader(props: Props) {
  const {
    event: { timestamp, name }
    // availableInventory,
    // totalInventory
  } = props;

  return (
    <Flex justify="space-between" align="flex-end">
      <FlexItem flex="0 0 auto" margin="0 auto 0 0">
        <EventTitle>{name}</EventTitle>
        <EventDate>{format(timestamp, DATE_FORMAT)}</EventDate>
        <EDLink to="#">
          {/* <InventoryLink color={cssConstants.PRIMARY_LIGHT_BLUE}>
            See Inventory List
          </InventoryLink> */}
        </EDLink>
      </FlexItem>
      {/* <FlexItem flex="0 0 auto">
        <P1 size="16px">Current Game Inventory</P1>
        <P1 size="22px" color={cssConstants.SECONDARY_BLUE}>
          {formatNumber(availableInventory)} / {formatNumber(totalInventory)}
        </P1>
        <ItalicS1 color={cssConstants.SECONDARY_BLUE}>
          Available / Total
        </ItalicS1>
      </FlexItem> */}
    </Flex>
  );
}
