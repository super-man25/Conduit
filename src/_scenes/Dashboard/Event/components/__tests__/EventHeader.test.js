import React from 'react';
import { shallow } from 'enzyme';
import { EventHeader } from '../EventHeader';
import { mockDateFnsFormat } from '_helpers/test-utils';

jest.mock('date-fns', () => ({
  format: (date, format) => mockDateFnsFormat(date, format)
}));

const props = {
  event: {
    clientId: 1,
    createdAt: 1529336865706,
    id: 1,
    integrationId: 1,
    modifiedAt: 1529336865706,
    name: 'Some Event',
    seasonId: 1,
    timestamp: 1529336865706,
    venueId: 1
  },
  availableInventory: 100,
  totalInventory: 1000
};

describe('<EventHeader />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<EventHeader {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
