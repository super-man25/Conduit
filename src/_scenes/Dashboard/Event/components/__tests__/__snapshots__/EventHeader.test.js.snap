// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventHeader /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventHeader
    availableInventory={100}
    event={
        Object {
            "clientId": 1,
            "createdAt": 1529336865706,
            "id": 1,
            "integrationId": 1,
            "modifiedAt": 1529336865706,
            "name": "Some Event",
            "seasonId": 1,
            "timestamp": 1529336865706,
            "venueId": 1,
          }
    }
    totalInventory={1000}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "align": "flex-end",
      "children": <FlexItem
        flex="0 0 auto"
        margin="0 auto 0 0"
>
        <styled.h1>
                Some Event
        </styled.h1>
        <styled.h4>
                2018-06-18T15:47:45.706Z @dddd, MMMM Do, YYYY @ h:mmA
        </styled.h4>
        <Styled(Link)
                to="#"
        />
</FlexItem>,
      "justify": "space-between",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.h1>
            Some Event
</styled.h1>,
          <styled.h4>
            2018-06-18T15:47:45.706Z @dddd, MMMM Do, YYYY @ h:mmA
</styled.h4>,
          <Styled(Link)
            to="#"
/>,
        ],
        "flex": "0 0 auto",
        "margin": "0 auto 0 0",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Some Event",
          },
          "ref": null,
          "rendered": "Some Event",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "2018-06-18T15:47:45.706Z @dddd, MMMM Do, YYYY @ h:mmA",
          },
          "ref": null,
          "rendered": "2018-06-18T15:47:45.706Z @dddd, MMMM Do, YYYY @ h:mmA",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "to": "#",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "flex-end",
        "children": <FlexItem
          flex="0 0 auto"
          margin="0 auto 0 0"
>
          <styled.h1>
                    Some Event
          </styled.h1>
          <styled.h4>
                    2018-06-18T15:47:45.706Z @dddd, MMMM Do, YYYY @ h:mmA
          </styled.h4>
          <Styled(Link)
                    to="#"
          />
</FlexItem>,
        "justify": "space-between",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.h1>
              Some Event
</styled.h1>,
            <styled.h4>
              2018-06-18T15:47:45.706Z @dddd, MMMM Do, YYYY @ h:mmA
</styled.h4>,
            <Styled(Link)
              to="#"
/>,
          ],
          "flex": "0 0 auto",
          "margin": "0 auto 0 0",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Some Event",
            },
            "ref": null,
            "rendered": "Some Event",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "2018-06-18T15:47:45.706Z @dddd, MMMM Do, YYYY @ h:mmA",
            },
            "ref": null,
            "rendered": "2018-06-18T15:47:45.706Z @dddd, MMMM Do, YYYY @ h:mmA",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "to": "#",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
